### DISCUSSION TITLE
Submitting a full 3D inflow plane data

### DISCUSSION THREAD

**Discussion by SRkumar97** [AMReX-Combustion/PeleC#882] [Q&A] [answered]
Hello! 
I'm trying to run a long domain, boundary layer case by splitting the simulation into parts. In the first part, the domain length would be only one half or one third of its actual length, while the next one will have the remaining portion. 
For this, the entire flow field data from the outlet plane of the first simulation should be fed to the inlet plane of the second simulation.
Essentially it would be something like exporting the entire flow snapshot of the x=x_hi plane of 3D simulation of domain 1, something like the exit plane contours of the images of this channel flow issue. And prescribe it for my domain 2 inlet. 
 https://github.com/AMReX-Combustion/PeleC/issues/263

 How do I go about this? Is this even possible?

**Comment by baperry2** [comment, marked as answer]
This capability is not available by default right now.  We do allow inflows that have spatially and temporally varying velocity read from a file (turbInflow), but right now that file is generated from an independent synthetic turbulence script. You could probably adapt that capability for your needs by extracting outflow data from your first simulation and writing it into the appropriate file format. This may be something that people have done, but hasn't made it back into the core code.

https://amrex-combustion.github.io/PelePhysics/Utility.html#turbulent-inflows

**Reply by SRkumar97** [reply]
Ohh, understood. Thanks! Do any of the test cases use this turbInflow feature?

**Reply by SRkumar97** [reply]
Found it. There is the homogeneous turbulence decay testcase https://amrex-combustion.github.io/PeleC/validation/validation.html#decay-of-homogeneous-isotropic-turbulence
Thanks!

**Reply by baperry2** [reply]
That case uses turbulence as an initial condition rather than a boundary condition. `Exec/Production/JetFlame` allows turbulence as a boundary condition. There's also a TurbInflow example in PeleLMeX that uses the same capability

**Reply by SRkumar97** [reply]
Ohh, thanks a lot. Will definitely check these.

**Reply by SRkumar97** [reply]
Hi, I am now trying to introduce a full laminar boundary layer (BL) profile at the inlet of the domain, in which I am specifying flow velocity and density in a functional form. Now given that the temperature also varies within the BL, I want to make use of the EOS() feature of PeleC instead of me fitting my data of temperature, in a functional form. My doubt whether such a use of EOS() is possible, since I am almost always seeing the use of EOS() in the testcases for regions with constant property i.e. single fixed value as in a freestream region. E.g. rho_in, T_in --> RYT2P(rho_in, massfrac, T_in, P_in). In my case I want the code to compute the Thermal BL based on my provided values for density and pressure in the bcnormal() function in prob.H. How to go about this?
Thanks in advance!

**Reply by SRkumar97** [reply]
Is this the correct way to do so, considering a case wherein I'd specify density profile in the **s_ext** array? Will this effectively read in the full arrays of density, mass frac and pressure at the inlet and thereby calculate temperature distribution and correctly map it to the entire **s_ext[UTEMP]** array?
eos.RYP2T(s_[URHO], massfrac, P_in, s_ext[UTEMP]);

**Reply by baperry2** [reply]
`bcnormal` operates on local data at a single cell. You can code an arbitrary functional form to specify density based on the position vector `x` of the cell that is also an argument to the function. 

`eos.RYP2T` is the right function to use if you have density, mass fractions, and pressure, and want to compute temperature.

Note that it is the user's responsibility to ensure you are specifying mathematically valid compressible flow boundary conditions if using bcnormal. if your inflow is subsonic, one flow characteristic leaves the domain at the inflow, so one flow variable should be specified based on the interior cells.

**Reply by SRkumar97** [reply]
Ohh, got it. thank you very much for the detailed info.

**Reply by SRkumar97** [reply]
> That case uses turbulence as an initial condition rather than a boundary condition. `Exec/Production/JetFlame` allows turbulence as a boundary condition. There's also a TurbInflow example in PeleLMeX that uses the same capability

Hi, I am trying to find the documentation or README file for this JetFlame case, but I'm unsure as to which of these flame cases listed in the documentation site corresponds to the JetFlame. Could someone please help?

**Reply by SRkumar97** [reply]
Also, how do we generate the turb.test2 file containing the fluctuations data, for this JetFlame case? In the PelePhysics documentation site, two methods are provided - Generating HIT file and generating inflow file
https://amrex-combustion.github.io/PelePhysics/Support.html#turbfilehit . 
I'm unaware as to how I should go about for generating the turb.test2 files as I couldn't find any README file for this JetFlame case. 
When I run the JetFlame case as is, the job exits with this error message on the Turb_test2/HDR/ files
- - - - 
>> amrex::Abort::227::Unable to open input file Turb.test2/HDR !!!
- - - -
Even if i manually edit the path to the Turb.test2/HDR/ folder in the input file, the same error still occurs.
- - - - - 
>> amrex::Abort::73::Unable to open input file /home/IITB/ag_aero/194010010/PeleC/Submodules/PelePhysics/Support/TurbFileHIT/Turb.test2/HDR !!!
- - - - - 

Requesting for help on this!

**Comment by d-montgomery** [comment]
Sorry for the lack of documentation on this case. Unfortunately once you get out of the `RegTests`, the documentation can be sparse.    The necessary DAT and HDR files are not included in the repo as they can be quite large.  Let me help walk you through the process of generating them:

1. From the PeleC directory, which I'll refer to as $PELE_C, navigate to the TurbFileHit directory in PelePhysics:
   ~~~
   cd $PELE_C/Submodules/PelePhysics/Support/TurbFileHIT
   ~~~
2. Follow the steps in [PelePhysics documentation](https://amrex-combustion.github.io/PelePhysics/Support.html#generating-an-hit-file) to generate the HIT files:
    * Run the python script to generate a `.dat` file.  Unfortunately, I can't find what `k0`, `Nk` were used in this case, so we'll use the defaults for those parameters, and we'll set `N = 64` which is the y-component of `amr.ncell` since the [jet is centered at (x,y) = (0,0)](https://github.com/AMReX-Combustion/PeleC/blob/343b8ef38fe6dfacde77de3db0afd097eddf91dd/Exec/Production/JetFlame/inputs.inp#L90) in the [z-direction](https://github.com/AMReX-Combustion/PeleC/blob/343b8ef38fe6dfacde77de3db0afd097eddf91dd/Exec/Production/JetFlame/inputs.inp#L86):
        ~~~
        ./gen_hit_ic.py -N 64
        ~~~
    * Next, compile the C++ source code that reads the `.dat` file created using the python script above and saves them as AMReX data structures that can be read by the TurbInflow utility:
      ~~~
      make -j
      ~~~
    * Run the C++ code:
       ~~~
       ./PeleTurb3d.gnu.ex input hit_file=hit_ic_4_64.dat input_ncell=64 TurbFile=Turb.test2
       ~~~
    * For this example, the required DAT and HDR files are saved in $PELE_C/Submodules/PelePhysics/Support/TurbFileHIT/Turb.test2
    * Move the Turb.test2 directory to the JetFlame case or whatever case you're trying to run.
       ~~~
       mv $PELE_C/Submodules/PelePhysics/Support/TurbFileHIT/Turb.test2 $PELE_C/Exec/Production/JetFlame
       ~~~

Now you should be able to run the case.

**Reply by SRkumar97** [reply]
Thanks a lot for this step-by-step procedure, I was able to generate all the files and run the JetFlame case!
